src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,210 @@
+
+    private static void eventRaid(final World raidingWorld, final int level, final Colony colony)
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = level;
+        int numberOfArcherBarbarians = (int) (ARCHER_BARBARIANS_MULTIPLIER * level);
+        int numberOfChiefBarbarians = (int) (CHIEF_BARBARIANS_MULTIPLIER * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > PREFERED_MAX_HORDE_SIZE && MAX_SIZE == PREFERED_MAX_HORDE_SIZE && hordeTotal > MAX_SIZE)
+        {
+            numberOfBarbarians = PREFERED_MAX_BARBARIANS;
+            numberOfArcherBarbarians = PREFERED_MAX_ARCHERS;
+            numberOfChiefBarbarians = PREFERED_MAX_CHIEFS;
+        }
+
+        if (hordeTotal > Configurations.maxBarbarianHordeSize)
+        {
+
+            numberOfBarbarians = hordeTotal - MAX_SIZE;
+
+            //For error handling and correct hordeTotal
+            if (numberOfBarbarians < 0)
+            {
+                numberOfBarbarians = 0;
+            }
+
+            //Re-asses hordeTotal to see if it is now below or at MAX_SIZE
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+            if (hordeTotal > MAX_SIZE)
+            {
+
+                numberOfArcherBarbarians = hordeTotal - MAX_SIZE;
+
+                //For error handling and correct hordeTotal
+                if (numberOfArcherBarbarians < 0)
+                {
+                    numberOfArcherBarbarians = 0;
+                }
+
+                //Re-asses hordeTotal to see if it is now below or at MAX_SIZE
+                hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+                if (hordeTotal > MAX_SIZE)
+                {
+
+                    numberOfChiefBarbarians = hordeTotal - MAX_SIZE;
+
+                    //For error handling and correct hordeTotal
+                    if (numberOfChiefBarbarians < 0)
+                    {
+                        numberOfChiefBarbarians = 0;
+                    }
+                }
+            }
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, colony, raidingWorld);
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final Colony colony, final World world)
+    {
+
+        final BlockPos targetSpawnPoint = calculateSpawnLocation(world, colony);
+
+        if (targetSpawnPoint != null && entityToSpawn != null && world != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)